File: C codes/bubbleSort.c
#include <stdio.h>
#include <stdlib.h>

void display(int arr[], int n)
{
    int i;
    for(i=0;i<n;i++)
    {
    printf("%d  ", arr[i]);
    }

}

void bubbleSort(int arr[], int n){
        int i,j,temp;
        int count=0;
        for( i=0;i<n-1;i++)
        {
            for(j=0;j<n-i-1;j++)
            {
                if(arr[j+1]<arr[j])
                {
                        temp = arr[j];
                        arr[j] = arr[j+1];
                        arr[j+1] = temp;
                }
                count++;
            }
            
        }
        printf("Array sorted after %d comparisons.", count);
        printf("Sorted Array: ");
        display(arr, n);
}

File: Java Fundamentals programs/BubbleSort.java
public class BubbleSort {

	public static void main(String[] args) {

		int[] intArray = { 20, 35, -15, 7, 55, 1, -22 };
		for (int lastUnsortedIndex = intArray.length - 1; lastUnsortedIndex > 0; lastUnsortedIndex--) {
			for (int i = 0; i < lastUnsortedIndex; i++) {
				if (intArray[i] > intArray[i + 1]) {
					swap(intArray, i, i + 1);
				}
			}
		}
		for (int i = 0; i < intArray.length; i++) {
			System.out.println(intArray[i]);
		}
	}

	public static void swap(int[] array, int i, int j) {

		if (i == j) {
			return;
		}
		int temp = array[i];
		array[i] = array[j];
		array[j] = temp;
	}

}
